### PR TITLE
Dbms-ordbms

### PR DESCRIPTION
**Pull request by Vaibhavpratapsingh22**
## Presenting 
Chatpter -7 **ORDBMS** 👍

### CODE REVIEW

File: 5-dbms/7-ordbms.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+### 2. MTI(Multi Table Inheritance)
+
+Multiple inheritance is a feature of some object-oriented computer programming languages in which an object or class can inherit characteristics and features from more than one parent object or parent class.
+
+We use MTI when the Subclasses have vastly different fields/columns but share common behavior. When subclasses have vastly different fields/columns, placing them all in a single table will have a lot of null columns, which is ultimately a waste of space, conceptually confusing, and reduces the effectiveness of indices. 
+
+<img src="https://danchak99.files.wordpress.com/2014/07/er_10061.jpg" height="" width="">
+
+## Collection Types
```
Comment (udeysingh95) [inline 5-dbms/7-ordbms.md:74]: Whats the difference between this collection types and the one - described in User-defined types.

File: 5-dbms/7-ordbms.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+### 1. Nested Tables
```
Comment (udeysingh95) [inline 5-dbms/7-ordbms.md:76]: Title after title

File: 5-dbms/7-ordbms.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+- Nested table is a table within a table.
```
Comment (udeysingh95) [inline 5-dbms/7-ordbms.md:77]: avoid direct points after titles.

File: 5-dbms/7-ordbms.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+- VARRAY can be created as either
+   -   an array of particular data type like NUMBER,  CHAR, VARCHAR, etc. 
+   - or an array of user defined data types.
+
+## Large Objects
+Large Objects(LOBs) are a collection of data-types which is designed to hold a large amount of data. A LOB is able to hold up to maximum size ranging from 8 terabytes to 128 terabytes, depending upon how your database is configured. Storing data into LOBs helps you to manipulate & access data efficiently in your own application. Large Objects are suitable for above last two kinds of data: semi-structured & unstructured data. Large objects features allow you to store these two kinds of data in the database as well as in operating system files that are usually accessed from the database.
+
+There are three types of LOBs as follow:
+
+### 1. BLOBs (Binary LOBs)
```
Comment (udeysingh95) [inline 5-dbms/7-ordbms.md:100]: Some times there are spaces before the bracket starts and sometimes there is not. Consistency improvement. Leave always one space before you start with a parenthesis.

File: 5-dbms/7-ordbms.md [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+   -   an array of particular data type like NUMBER,  CHAR, VARCHAR, etc. 
+   - or an array of user defined data types.
+
+## Large Objects
+Large Objects(LOBs) are a collection of data-types which is designed to hold a large amount of data. A LOB is able to hold up to maximum size ranging from 8 terabytes to 128 terabytes, depending upon how your database is configured. Storing data into LOBs helps you to manipulate & access data efficiently in your own application. Large Objects are suitable for above last two kinds of data: semi-structured & unstructured data. Large objects features allow you to store these two kinds of data in the database as well as in operating system files that are usually accessed from the database.
+
+There are three types of LOBs as follow:
+
+### 1. BLOBs (Binary LOBs) 
+A Binary Large OBject (BLOB) is a collection of binary data stored as a single entity in a database management system. Blobs are typically images, audio or other multimedia objects, though sometimes binary executable code is stored as a blob. Database support for blobs is not universal.
```
Comment (udeysingh95) [inline 5-dbms/7-ordbms.md:101]: Blobs are very very important in modern DBs - and you have covered it nicely.

**Review by udeysingh95 — COMMENTED** [review]
Overall very good work. 
I think all the topics are covered nicely - also the sizing and language are very appropriate in this PR. I really liked the first version of work this time. Keep it up. 
There are some minor suggested changes - once done - its good to go from my side.

**Comment by udeysingh95** [comment]
Very good images.

File: 5-dbms/7-ordbms.md (hunk 1 above)
Comment (Vaibhavpratapsingh22) [reply 5-dbms/7-ordbms.md:74]: Collection type is just like storing arrays whereas user-defined data type is a special feature of an object-oriented system. UDT allows user to define a data type of the data and how should it work.

**Review by udeysingh95 — APPROVED** [review]

File: 5-dbms/7-ordbms.md [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+It is similar to a binary large object (BLOB) in that they are both large amounts of data. The key difference is that a CLOB is stored using a text encoding method such as ASCII or Unicode. CLOBs can encompass long documents such as books. They can range up to two gigabytes in size or even more. Some databases store CLOBs by referencing out-of-table data.
+
+### 3. BFILE Datatype
+External LOBs are data objects stored in operating system files outside the database table-spaces, that have no transactional support from the database server.
+BFILEs are having read-only datatypes. The database allows read-only byte stream access to data stored in BFILEs. You cannot write to a BFILE from within your application.
+
+
+## Summary
+
+Above we learned about ORDBMS and its functions or we can say features like Row & Reference types,User-defined types, Supertype / Subtype relationships, User-defined procedures, Table Inheritance, Collection Types and Large objects.
```
Comment (SaarthakDeshmukh) [inline 5-dbms/7-ordbms.md:114]:
```suggestion
Above we learned about ORDBMS and its functions or we can say features like Row & Reference types, User-defined types, Supertype / Subtype relationships, User-defined procedures, Table Inheritance, Collection Types and Large objects.
```

File: 5-dbms/7-ordbms.md [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+- Nested tables are unbounded, ie., it can grow in size. In other words, we don’t specify the maximum size as we do for other data types.
+- Within the database, nested tables can be considered one-column database tables. Oracle stores the rows of a nested table in no particular order. But, when you retrieve the nested table into a PL/SQL variable, the rows are given consecutive subscripts starting at 1. That gives you array-like access to individual rows.
+- A nested table can be either used in a PL/SQL block to declare as a variable temporarily or as a column of a table which can persistently store the data in the database. 
+
+### 2. VARRAY (Varrying Array)
+
+VARRAY is one of the collection data types in Oracle to store column(s) that contain multiple values. That is, an attribute of type VARRAY is a multi-valued attribute. For example, assume a customer with multiple phone numbers. To store all phone numbers as part of single record, we would use VARRAY. Some properties of VARRAY are: 
+- Unlike nested tables, VARRAY has a limit. In other words, we use VARRAY attributes to store limited number of entries. It has a limit like other data types NUMBER, CHAR etc.
```
Comment (SaarthakDeshmukh) [inline 5-dbms/7-ordbms.md:88]: Good example here, I could relate to the topic.

File: 5-dbms/7-ordbms.md [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+
+<img src="https://danchak99.files.wordpress.com/2014/07/er_10061.jpg" height="" width="">
+
+## Collection Types
+Collection types are the common variations of data collections, such as:
+
+### 1. Nested Tables
+ An ordered group of items of type TABLE are called nested tables. Nested table is a table within a table and some of its properties are:
+
+- Nested tables can contain multiple columns.
```
Comment (SaarthakDeshmukh) [inline 5-dbms/7-ordbms.md:80]: Good work here, the points are easy to understand, But always remember BULLET points should not end with a full stop (.) Change it here as well as in the below sections.

File: 5-dbms/7-ordbms.md [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+
+There are two types of table inheritance as follow:
+
+## 1. STI (Single Table Inheritance)
+
+Single table inheritance is a way to emulate object-oriented inheritance in a object-relational database. When mapping from a database table to an object in an object-oriented language, a field in the database identifies what class in the hierarchy the object belongs to. All fields of all the classes are stored in the same table, hence the name "Single Table Inheritance". 
+
+We use STI when the Subclasses have the same fields/columns but different behavior. A good indication, that STI is right is when the different subclasses have the same fields/columns but different methods. 
+
+<img src="https://cayenne.apache.org/docs/3.0/images/single-table-inheritance.png" height="" width="">
```
Comment (SaarthakDeshmukh) [inline 5-dbms/7-ordbms.md:64]: Images can be placed in the center for better readability.

File: 5-dbms/7-ordbms.md [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+
+## Table Inheritance
+ Tables that are defined on named row types support table inheritance. Table inheritance is the property that allows a table to inherit the behavior (constraints, storage options, triggers) from the super-table above it in the table hierarchy. A table hierarchy is the relationship that you can define among tables in which sub-tables inherit the behavior of super-tables. A table inheritance provides the following advantages:
+It encourages modular implementation of your data model.
+- It ensures consistent reuse of schema components.
+- It allows you to construct queries whose scope can be some or all of the tables in the table hierarchy.
+
+There are two types of table inheritance as follow:
+
+## 1. STI (Single Table Inheritance)
```
Comment (SaarthakDeshmukh) [inline 5-dbms/7-ordbms.md:58]: Check out the levels here. I think this should be on level-3

File: 5-dbms/7-ordbms.md [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+
+A supertype is a generic entity type that has a relationship with one or more subtypes. A subtype is a sub-grouping of the entities in an entity type that is meaningful to the organization and that shares common attributes or relationships distinct from other subgroups.
+
+There are two criteria that help the designer determine when to use subtypes and super-types:
+
+1. There must be different, identifiable kinds or types of the entity in the user’s environment.
+
+2. The different kinds or types of instances should have one or more attributes that are unique to that kind or type of instance.
+
+<img src="http://www.myreadingroom.co.in/images/stories/docs/dbms/Entity%20Supertypes%20and%20Subtypes.JPG" height="" width="" alt="Sorry, Network Error">
```
Comment (SaarthakDeshmukh) [inline 5-dbms/7-ordbms.md:48]: I couldn't easily read the contents inside the image. Quality can be improved

**Review by SaarthakDeshmukh — CHANGES_REQUESTED** [review]
The Quality of the Chapter is really nice. I could brush up my knowledge and it was keeping me connected to the topic. That was really nice. There are some changes from my side to improve readability and few structure related changes .

**Review by udeysingh95 — APPROVED** [review]

**Review by SaarthakDeshmukh — APPROVED** [review]
Good Work @Vaibhavpratapsingh22 Happy to see you progressing.

**Pull request merged**